### PR TITLE
speed up symmetric_difference

### PR DESCRIPTION
**Pull request by alimamdouh212**
```g1 = path_digraph(100_000)
g2 = SimpleDiGraph(1)

@benchmark symmetric_difference2(g1, g2)
BenchmarkTools.Trial: 
  memory estimate:  11.06 MiB
  allocs estimate:  200003
  --------------
  minimum time:     22.693 ms (0.00% GC)
  median time:      32.740 ms (0.00% GC)
  mean time:        33.670 ms (11.81% GC)
  maximum time:     60.221 ms (41.67% GC)
  --------------
  samples:          149
  evals/sample:     1

@benchmark symmetric_difference(g1, g2)
BenchmarkTools.Trial: 
  memory estimate:  22.13 MiB
  allocs estimate:  400003
  --------------
  minimum time:     41.657 ms (0.00% GC)
  median time:      69.306 ms (0.00% GC)
  mean time:        70.907 ms (16.42% GC)
  maximum time:     107.075 ms (36.01% GC)
  --------------
  samples:          71
  evals/sample:     1
```

### CODE REVIEW

**Comment by sbromberger** [comment]
Please stop submitting PRs that aren't on target per #1366.

**Comment by alimamdouh212** [comment]
operator doesn’t have 13 suffix, so i thought i can submit it

**Comment by simonschoelly** [comment]
Still, as you can see by the many open prs we don't have enough time to review them all, so please don't open anymore, until we have merged at least a substantional amount of the ones we have already here.

**Comment by sbromberger** [comment]
@alimamdouh212 please stop. Every time you make a new PR or push to an existing PR we get notified. It's become quite a distraction as we're really focusing on our 1.4.0/2.0.0 efforts right now. Over the past 2 days I have over 40 emails from your activity alone.

I appreciate your efforts, but for now please don't create new PRs unless they follow the guidelines set out in #1366 , and don't make any changes to existing PRs. If you want to queue up some work, please do so in your own local repo and we'll get to it after we slow down on our current code drive.

**Comment by alimamdouh212** [comment]
is n’t files with no 13 suffix ok ?

**Comment by sbromberger** [comment]
> is n’t files with no 13 suffix ok ?

No. The point is that unless there's a critical bug fix required, nothing that's not related to the new 1.4.0/2.0.0 code is going to be reviewed in the near term. We're focused on that. I've edited #1366 to make it more clear.

File: test/operators.jl [hunk 1]
```diff
@@ -38,6 +38,16 @@
             @test ne(z) == 1
         end
 
+        @testset "symmetric difference for Digraph" begin
+            g1 = path_digraph(2)
+            g2 = path_digraph(4)
+
+            y = symmetric_difference2(g1, g2)
+            @test y == symmetric_difference2(g2, g1)
```
Comment (sbromberger) [inline test/operators.jl:46]: where is this function defined? (Is it left over from testing?)

Comment (alimamdouh212) [reply test/operators.jl]: it is a mistake.

Comment (sbromberger) [reply test/operators.jl]: OK, Cool. Just FYI, things are slowing down on the 2.0.0 changes so I'm ready to review the backlog. Thanks for being so patient.

**Comment by sbromberger** [comment]
I think this is ok to merge now. Thank you, @alimamdouh212 !

File: src/operators.jl [hunk 2]
```diff
@@ -236,20 +236,119 @@ julia> collect(edges(f))
  Edge 2 => 3
 ```
 """
-function symmetric_difference(g::T, h::T) where T <: AbstractGraph
-    gnv = nv(g)
-    hnv = nv(h)
+function symmetric_difference(g::T, h::T) where T <: SimpleDiGraph
```
Comment (sbromberger) [inline src/operators.jl:239]: Oh, wait. This should be:
```
@traitfn function symmetric_difference(g::::IsDirected, h::::IsDirected)
```

and the one for undirected graphs should be `(!IsDirected)`.

Comment (sbromberger) [reply src/operators.jl]: I forgot that that syntax doesn't work, but I think you got the correct one.

**Comment by sbromberger** [comment]
Can you figure out a test that exercises https://codecov.io/gh/JuliaGraphs/LightGraphs.jl/src/9480b512446ebce7f912d80f8b1807de16267f43/src/operators.jl#L333 ?

**Comment by sbromberger** [comment]
Thanks!

**Pull request merged**